File: CouponSystem/Java/java/com/coupon/beans/Customer.java
package com.coupon.beans;

import java.io.Serializable;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.xml.bind.annotation.XmlRootElement;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;

import com.fasterxml.jackson.annotation.ObjectIdGenerators;

/**
 * Company Profile
 * 
 * @author Musab
 * 
 */
@SuppressWarnings("serial")
@Entity(name = "Customer")
@XmlRootElement
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Customer implements Serializable
{
	@Id
	@Column(name = "ID", unique = true, nullable = false)
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int ID;
	@Column(name = "CUST_NAME", unique = true)
	private String CUST_NAME;
	@Column(name = "PASSWORD")
	private String PASSWORD;
	@ManyToMany(mappedBy = "customers")
	private List<Coupon> coupons;

	public Customer()
	{
		// TODO Auto-generated constructor stub
	}

	public Customer(int iD, String cUST_NAME, String pASSWORD, List<Coupon> coupons)
	{
		super();
		ID = iD;
		CUST_NAME = cUST_NAME;
		PASSWORD = pASSWORD;
		this.coupons = coupons;
	}

	public int getID()
	{
		return ID;
	}

	public void setID(int iD)
	{
		ID = iD;
	}

	public String getCUST_NAME()
	{
		return CUST_NAME;
	}

	public void setCUST_NAME(String cUST_NAME)
	{
		CUST_NAME = cUST_NAME;
	}

	public String getPASSWORD()
	{
		return PASSWORD;
	}

	public void setPASSWORD(String pASSWORD)
	{
		PASSWORD = pASSWORD;
	}

	public List<Coupon> getCoupons()
	{
		return coupons;
	}

	public void setCoupons(List<Coupon> coupons)
	{
		this.coupons = coupons;
	}

	@Override
	public String toString()
	{
		return "Customer [ID=" + ID + ", CUST_NAME=" + CUST_NAME + ", PASSWORD=" + PASSWORD + ", coupons=" + coupons + "]";
	}
}

File: CouponSystem/Angular/CustomerScreen/src/app/Beans/Customer.ts
import { Coupon } from '../Beans/Coupon' ;

export class Customer {
    constructor(private _id?:number, private _cust_NAME?:string,
                 private _password?:string, private _coupons?:Coupon[])
    {
    }
    get id() : number {
      return this._id;
    }
    set id(newID : number)
    {
      this._id = newID;
    }
    get cust_NAME() : string {
        return this._cust_NAME;
      }
      set cust_NAME(newName : string)
      {
        this._cust_NAME = newName;
      }
      get password() : string {
        return this._password;
      }
      set password(newName : string)
      {
        this._password = newName;
      }
   
      get coupons() : Coupon[] {
        return this._coupons;
      }
      set coupons(newName : Coupon[])
      {
        this._coupons = newName;
      }
   
  }
  
